cachi2/core/package_managers/yarn_classic/main.py

```diff
@@ -16,3 +81,27 @@ def _generate_build_environment_variables() -> list[EnvironmentVariable]:
     }
 
     return [EnvironmentVariable(name=key, value=value) for key, value in env_vars.items()]
+
+
+def _verify_corepack_yarn_version(source_dir: RootedPath, env: dict[str, str]) -> None:
+    """Verify that corepack installed the correct version of yarn by checking `yarn --version`."""
+    yarn_version_output = run_yarn_cmd(["--version"], source_dir, env=env).strip()
+
+    try:
+        installed_yarn_version = semver.version.Version.parse(yarn_version_output)
+    except ValueError as e:
+        raise PackageManagerError(
+            "Cachi2 expected corepack to install yarn >=1.22.0,<2.0.0, but "
```

Comment:
Does the version check actually happen here? It looks like the only thing we can be sure about at this point is that `The command 'yarn --version' did not return a valid semver.")

Reply:
Good point, updated :+1: